core/src/main/java/org/elasticsearch/common/geo/GeoUtils.java

```diff
@@ -56,6 +65,30 @@
     /** Earth ellipsoid polar distance in meters */
     public static final double EARTH_POLAR_DISTANCE = Math.PI * EARTH_SEMI_MINOR_AXIS;
 
+    /** Returns true if value is neither NaN nor infinite. */
+    public static boolean isValidNumber(double value) {
```

Comment:
that is a different story :) I don't know where to move it either, would it be possible to make it package private at least?